### PR TITLE
Improves formatting of docstring + newlines

### PR DESCRIPTION
**Pull request by northern-64bit**
# What does this PR do?
Improve docstrings and kind of standardize them (allowing some variations). It further also makes sure that there is 2 empty rows after and before classes or functions (if they are not a part of a class), and otherwise not.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by kashif** [comment]
@northern-64bit there might be some potential merge conflicts when  #1944 is merged

**Comment by northern-64bit** [comment]
> @northern-64bit there might be some potential merge conflicts when #1944 is merged

Thanks for letting me know. I've now fixed it.

**Comment by qgallouedec** [comment]
Thanks a lot for cleaning the doc! Please address my comment and we're ready to merge :)

File: trl/trainer/cpo_config.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class CPOConfig(TrainingArguments):
     [argparse](https://docs.python.org/3/library/argparse#module-argparse) arguments that can be specified on the
     command line.
 
-    Parameters:
+    Args:
```
Comment (qgallouedec) [inline trl/trainer/cpo_config.py:29]:
```suggestion
    Parameters:
```

For configs (only) I'd keep `Parameters` to be consistent with `TrainingArguments`. Even though I agree that `Args` is more aligned with the convention.

Comment (northern-64bit) [reply trl/trainer/cpo_config.py:29]: Sounds good, I'll change it then.

Comment (northern-64bit) [reply trl/trainer/cpo_config.py]: Done :)

File: trl/trainer/utils.py [hunk 2]
```diff
@@ -897,7 +900,7 @@ class OnPolicyConfig(TrainingArguments):
     [argparse](https://docs.python.org/3/library/argparse#module-argparse) arguments that can be specified on the
     command line.
 
-    Parameters:
+    Args:
```
Comment (qgallouedec) [inline trl/trainer/utils.py:903]:
```suggestion
    Parameters:
```

File: trl/trainer/utils.py [hunk 3]
```diff
@@ -450,32 +452,32 @@ class ConstantLengthDataset(IterableDataset):
...
-            append_concat_token (`bool`, *optional*, defaults to True)
-                If true, appends `eos_token_id` at the end of each sample being packed.
-            add_special_tokens (`bool`, *optional*, defaults to True)
-                If true, tokenizers adds special tokens to each sample being packed.
+    Args:
+        tokenizer (`transformers.PreTrainedTokenizer`):
+            The processor used for processing the data.
+        dataset (`dataset.Dataset`):
+            Dataset with text files.
+        dataset_text_field (`str`, **optional**):
```
Comment (qgallouedec) [inline trl/trainer/utils.py:460]: Mention the default to None please

File: trl/trainer/utils.py [hunk 4]
```diff
@@ -450,32 +452,32 @@ class ConstantLengthDataset(IterableDataset):
...
+            Length of token sequences to return.
+        num_of_sequences (`int`, *optional*, defaults to `1024`):
+            Number of token sequences to keep in buffer.
+        chars_per_token (`int`, *optional*, defaults to `3.6`):
+            Number of characters per token used to estimate number of tokens in text buffer.
+        eos_token_id (`int`, *optional*, defaults to `0`):
+            Id of the end of sequence token if the passed tokenizer does not have an EOS token.
+        shuffle (`bool`, *optional*, defaults to True)
+            Shuffle the examples before they are returned
+        append_concat_token (`bool`, *optional*, defaults to True)
```
Comment (qgallouedec) [inline trl/trainer/utils.py:477]: Use ` for default values

File: trl/trainer/alignprop_trainer.py [hunk 5]
```diff
@@ -56,12 +55,16 @@ class AlignPropTrainer(BaseTrainer):
...
     Attributes:
-        **config** (`AlignPropConfig`) -- Configuration object for AlignPropTrainer. Check the documentation of `PPOConfig` for more
-         details.
-        **reward_function** (Callable[[torch.Tensor, Tuple[str], Tuple[Any]], torch.Tensor]) -- Reward function to be used
-        **prompt_function** (Callable[[], Tuple[str, Any]]) -- Function to generate prompts to guide model
-        **sd_pipeline** (`DDPOStableDiffusionPipeline`) -- Stable Diffusion pipeline to be used for training.
-        **image_samples_hook** (Optional[Callable[[Any, Any, Any], Any]]) -- Hook to be called to log images
+        config (`AlignPropConfig`):
+            Configuration object for AlignPropTrainer. Check the documentation of `PPOConfig` for more details.
+        reward_function (Callable[[torch.Tensor, Tuple[str], Tuple[Any]], torch.Tensor]):
```
Comment (qgallouedec) [inline trl/trainer/alignprop_trainer.py:60]: Use ˋ for type

**Review by qgallouedec — CHANGES_REQUESTED** [review]

**Review by qgallouedec — APPROVED** [review]

**Pull request merged**